protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/FetchProcessor.java

```diff
@@ -102,7 +107,7 @@ protected void processRequest(FetchRequest request, ImapSession session, Respond
             if (vanished) {
```

Comment:
Above...

`boolean vanished = fetch.getVanished();`

This is the vanished condition I mention.